extension-ui/background.js

```diff
@@ -193,3 +193,198 @@ function showNotification(site, num, options = {}) {
   
   chrome.notifications.create(notificationOptions);
 }
+
+let timeoutId = null;
+let notificationTimeoutId = null;
+let remainingTime = initialTime;
+let lastVisitTime = Date.now();
+let lastVisitDate = new Date().toDateString();
+let isSiteBlocked = false;
+let notificationShown = false;
+let currentTabId = null;
+let currentTabUrl = null;
+let isTabActive = false; 
+
+function updateStorage() {
+  chrome.storage.local.set({
+    lastVisitDate,
+    remainingTime,
+    isSiteBlocked,
+    notificationShown
+  });
+}
+
+function checkForNewDay() {
+  const currentDate = new Date().toDateString();
+  if (currentDate !== lastVisitDate) {
+    remainingTime = initialTime;
+    lastVisitDate = currentDate;
+    isSiteBlocked = false;
+    notificationShown = false;
+    updateStorage();
+  }
+}
+
+function startTimer() {
+  checkForNewDay();
```

Comment:
you need to check also if the active profile was changed since the last block
(for example if in work profile the site is limited ti 2 hours, and in home profile the site is limited to 3 hours, so he has 5 hours a day)